doc/aaa/TACACS+ Design.md

```diff
@@ -0,0 +1,537 @@
+- [7 Unit Test](#unit-test)
+  * [7.1 Unit test for source code](#71-unit-test-for-source-code)
+  * [7.2 End to end test with testbed](#72-end-to-end-test-with-testbed)
+  * [7.3 Backward compatibility test](#73-backward compatibility test)
+- [8 References](#references)
+  * [RFC8907](#rfc8907)
+  * [TACACS+ Authentication](#tacacs+-Authentication)
+  * [Bash](#bash)
+  * [pam_tacplus](#pam_tacplus)
+  * [Auditd](#auditd)
+  * [audisp-tacplus](#audisp-tacplus)
+  * [tacplus-auth](#tacplus-auth)
+
+# About this Manual
+This document is based on [TACACS+ Authentication](#TACPLUS-Authentication), and provides a detailed description on the new features for improve TACACS+ support.
+
+ - SONiC currently supported TACACS+ features:
+   - Authentication.
+   - User session authorization.
+   - User session accounting.
+   - User command authorization with local permission.
+
+ - New features:
+   - User command authorization with TACACS+ server.
+   - User command accounting with TACACS+ server.
+
+# 1 Functional Requirement
+## 1.1 User command authorization
+ - Authorization when user run any executable file or script on SONiC host.
+   - The full path and parameters will be send to TACACS+ server side for authorization.
+   - For recursive command/script, only the top level command have authorization.
+   - No authorization for  bash built-in command and bash function, but if a bash function call any executable file or script, those executable file or script will have authorization.
+
+ - TACACS+ authorization is configurable:
+   - TACACS+ authorization can be enable/disable.
+   - TACACS+ authorization method will communicate with TACACS+ server for authorization, TACACS+ server should setup permit/deny rules.
+
+ - Failover:
+   - If a TACACS+ server not accessible, the next TACACS+ server authorization will be performed.
+   - When all remote TACACS+ server not accessible, TACACS+ authorization will failed.
+   - When TACACS+ authorization failed, fallover is configurable:
+     - Disable local authorization as failover, then user can't run any command.
+     - Enable local authorization as failover, then user can run command with local authorization.
+     - For local authorization, please check [TACACS+ Authentication](#TACPLUS-Authentication).
+   - When user login with local account, TACACS+ authentication and authorization will disabled for current user.
+     - After login, user can run command with local authorization.
+     - When all TACACS+ server not accessible, user can login with this method.
+## 1.2 User command accounting
+ - Accounting is the action of recording what a user is doing, and/or has done.
+
+ - Following event will be accounted:
+   - Command start event.
+   - Command finish event.
+
+ - User command in Docker will not be accounted.
+   - User command in docker actually run by docker service, so we can't identify if command are run by user or system service.
```

Comment:
>docker [](http://example.com/codeflow?start=21&length=6)

I guess you mean shell environment inside docker container. On the other hand, the `docker exec <container> <process>` itself will be accounted.

Reply:
Fixed.